src/dynamics/video_player/player/player.js

```diff
@@ -2657,11 +2678,27 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                                     return promise.asyncError(error);
                                 }, this) :
                                 console.log("Please define requestForTheNextAdTagURL method with Promise.");
-                        }, this, immediate ? 100 : this.get("outstreamoptions.recurrenceperiod") || 30000);
+                        }, this, immediate ? 100 : requestInterval || 30000);
                     }
                     return promise;
                 },
 
+                resetOutstreamRetries() {
+                    this.set("maxRetriesMet", false);
+                    this.set("retriesOnError", 0);
```

Comment:
nitpick: can't you handle this with 1 variable? like create a variable such that`retriesAvailable:0` means that we have no retries left?

Reply:
sure I can do that!